roles/st2/tasks/auth.yml

```diff
@@ -39,3 +39,18 @@
   notify:
     - restart st2api
     - restart st2stream
+
+- name: auth | Create root's CLI configuration directory
+  file: path=/root/.st2 state=directory
+  when: st2_save_credentials
+
+- name: auth | Save credentials in CLI configuration file
+  become: yes
+  blockinfile:
+    dest: /root/.st2/config
+    create: yes
+    block: |
+      [credentials]
+      username = {{ st2_auth_username }}
+      password = {{ st2_auth_password }}
```

Comment:
For security reasons please set explicitly `mode` so the file with auth credentials will be readable only by `root` user.

In addition to that, just to be sure it would be nice to include `owner` and `group` settings as well.
https://docs.ansible.com/ansible/blockinfile_module.html